### DISCUSSION TITLE
Dynamic site Url

### DISCUSSION THREAD

**Discussion by Erildo** [supabase/supabase#8738] [Questions]
I have uploaded the site to vercel, the domain has changed from localhost to myapp.vecerl.com

I keep changing the site url to that of localhost whenever I am testing, how can I make it to detect in which site I am and change the site url to one of those that I have in redirect url?
![image](https://user-images.githubusercontent.com/3402855/187636021-8644ad80-773f-489b-9384-a2c94976a788.png)

**Comment by GaryAustin1** [comment]
There is no automatic way to change the site URL.
Your redirect list should allow the redirectTo parameter (set with the appropriate domain by client code) to go thru.
The site URL is the default test if not found in the redirect list, and for informative purposes in the standard templates.
What are you using the site URL for that you need to change it for dev?

Warning I have medium confidence in this answer as there are lots of different discussions/issues around it.  I've never changed my site URL when testing on various localhost ports and on my webserver, The only place I see it show up in a standard template is the greeting: `You have been invited to create a user on {{ .SiteURL }}`

**Reply by Erildo** [reply]
Hi Gary,
I need to test the features that I am developing then push them to Prod.
Since I am using Linkedin Provider I need the session Auth provided from Linkedin within the webApp to get further information and that redirects to the site Url...Supabase needs somehow to detect the site where I am and then change it accordingly or I need to find another solution to it...

**Comment by grieblius** [comment]
From my experience you do not need to  change the Site URL every time.
What you need is to add exact urls in "Redirect URLs" sections with other domain.
E.g. if after login you redirect to `http://localhost:1234/account`, then you need to add it to "Redirect URLs".

Also, you need to use `redirectTo` option with `signIn` method.
Docs: [https://supabase.com/docs/reference/javascript/auth-signin](https://supabase.com/docs/reference/javascript/auth-signin)

`redirectTo` should be full url, including domain. And with other than Site Url domain it will work only if the exact url is added to "Redirect URLs" section. E.g. if added `http://localhost:1234/account` and you pass it with `redirectTo` - it should work.

UPDATE: 
Just found, it is not necessary to define exact URL in "Redirect URLS", now we can use a wildcard e.g. `http://localhost:1234/*`:
https://github.com/supabase/supabase/discussions/600#discussioncomment-3303368

**Reply by azashi** [reply]
One more suggestion - adding urls like ``` http://localhost:1234/**/** ``` also makes sure any trailing slash and nested paths are redirected as well. 
Hope this helps!
